galasa-extensions-parent/dev.galasa.extensions.common/src/main/java/dev/galasa/extensions/common/couchdb/CouchdbStore.java

```diff
@@ -125,6 +130,24 @@ protected <T> T getDocumentFromDatabase(String dbName, String documentId, Class<
         return gson.fromJson(sendHttpRequest(getDocumentRequest, HttpStatus.SC_OK), classOfObject);
     }
 
+
+    protected void retrieveArtifactFromDatabase(String URI, Path cachePath, CopyOption copyOption) throws CouchdbException{
+        HttpGet httpGet = httpRequestFactory.getHttpGetRequest(URI);
+        try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
+            StatusLine statusLine = response.getStatusLine();
+            if (statusLine.getStatusCode() != HttpStatus.SC_OK) {
+                String errorMessage = ERROR_URI_IS_INVALID .getMessage(URI);
+                throw new CouchdbException(errorMessage);
+            }
+            HttpEntity entity = response.getEntity();
+            Files.copy(entity.getContent(), cachePath, copyOption);
```

Comment:
Would be good to replace `Files.copy` with a filesystem that can be mocked out for unit testing (not sure if we already have any unit tests for this!)